pkg/apis/ingressnginxplugin/v1alpha1/types.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressNginxPluginis a top-level type
+type IngressNginxPlugin struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
+	// Status HelloTypeStatus `json:"status,omitempty"`
+	// This is where you can define
+	// your own custom spec
+	Spec IngressNginxPluginSpec `json:"spec,omitempty"`
+}
+
+// custom spec
+type IngressNginxPluginSpec struct {
```

Comment:
I don't see the point of a CRD for this definition. If we are just going to store base64 definition of the plugin, then using a configmap is a more proper solution than a custom CRD